crates/bevy_math/src/common_traits.rs

```diff
@@ -0,0 +1,133 @@
+use glam::{Quat, Vec2, Vec3, Vec3A, Vec4};
+use std::fmt::Debug;
+use std::ops::{Add, Div, Mul, Sub};
+
+/// A type that supports the mathematical operations of a vector space, irrespective of dimension.
+pub trait VectorSpace:
+    Mul<f32, Output = Self>
+    + Div<f32, Output = Self>
+    + Add<Self, Output = Self>
+    + Sub<Self, Output = Self>
+    + Default
+    + Debug
+    + Clone
+    + Copy
+{
+    /// Perform vector space linear interpolation between this element and another, based
+    /// on the parameter `t`. When `t` is `0`, `self` is recovered. When `t` is `1`, `rhs`
+    /// is recovered.
+    ///
+    /// Note that the value of `t` is not clamped by this function, so interpolating outside
+    /// of the interval `[0,1]` is allowed.
+    #[inline]
+    fn lerp(&self, rhs: Self, t: f32) -> Self {
+        *self * (1. - t) + rhs * t
+    }
+}
+
+impl VectorSpace for Quat {}
+impl VectorSpace for Vec4 {}
+impl VectorSpace for Vec3 {}
+impl VectorSpace for Vec3A {}
+impl VectorSpace for Vec2 {}
+impl VectorSpace for f32 {}
+
+/// A type that supports the operations of a normed vector space; i.e. a norm operation in addition
+/// to those of [`VectorSpace`]. The implementor must guarantee that the axioms of a normed vector
+/// space are satisfied.
+pub trait NormedVectorSpace: VectorSpace {
```

Comment:
The mathematician in me also wants to see `MetricSpace` and `InnerProductSpace` traits with corresponding blanket impls, but I agree with Alice that we shouldn't add traits until the need arises.

Reply:
I agree! I suspect it will come up before too long if someone tries to use `NormedVectorSpace` and ends up needing a dot product.